### PR TITLE
feat: add to cache store files

### PR DESCRIPTION
**Pull request by J1za**
## Linear Ticket
https://linear.app/autokitteh/issue/UI-923/cache-files
## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/store/cache/useCacheStore.ts [hunk 1]
```diff
@@ -69,13 +74,64 @@ const store: StateCreator<CacheStore> = (set, get) => ({
...
+			const { data, error } = await ProjectsService.getResources(projectId);
+
+			if (error) {
+				return;
+			}
+			await dbService.clearStore();
+			for (const [name, content] of Object.entries(data || {})) {
+				await dbService.put(name, new Uint8Array(content));
+			}
+			const resources = await dbService.getAll();
```
Comment (RonenMars) [inline src/store/cache/useCacheStore.ts:109]: Let's reduce this db query:
```suggestion
			const resources = data;
```

Comment (J1za) [reply src/store/cache/useCacheStore.ts:109]: refactored

**Review by RonenMars — COMMENTED** [review]
Good job!
good job

File: src/store/cache/useCacheStore.ts [hunk 2]
```diff
@@ -66,13 +71,64 @@ const store: StateCreator<CacheStore> = (set, get) => ({
...
+
+		set((state) => ({
+			...state,
+			loading: { ...state.loading, resourses: true },
+		}));
+		try {
+			const { data, error } = await ProjectsService.getResources(projectId);
+
+			if (error) {
+				return;
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/store/cache/useCacheStore.ts:100]: added

File: src/hooks/useFileOperations.ts [hunk 3]
```diff
@@ -145,16 +98,13 @@ export function useFileOperations(projectId: string) {
 		saveFile,
 		saveAllFiles,
 		deleteFile,
-		fetchResources,
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/hooks/useFileOperations.ts:148]: added

File: src/store/cache/useCacheStore.ts [hunk 4]
```diff
@@ -66,13 +71,64 @@ const store: StateCreator<CacheStore> = (set, get) => ({
...
+		if (currentProjectId === projectId && !force) {
+			return resources;
+		}
+
+		set((state) => ({
+			...state,
+			loading: { ...state.loading, resourses: true },
+		}));
+		try {
+			const { data, error } = await ProjectsService.getResources(projectId);
```
[the remark this answers is not in the document]
Comment (J1za) [reply src/store/cache/useCacheStore.ts:97]: changed

**Review by RonenMars — APPROVED** [review]

**Pull request merged**